waku/v2/protocol/rln/waku_rln_relay.go

```diff
@@ -69,31 +50,39 @@ func New(
 		Path:          treePath,
 	})
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
-	metrics.RecordInstanceCreation(time.Since(start))
 
 	rootTracker, err := group_manager.NewMerkleRootTracker(acceptableRootWindowSize, rlnInstance)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
+	return rlnInstance, rootTracker, nil
+}
+func New(
+	gmDetails group_manager.GMDetails,
+	timesource timesource.Timesource,
+	reg prometheus.Registerer,
+	log *zap.Logger) *WakuRLNRelay {
+
+	start := time.Now()
+	metrics := newMetrics(reg)
+	metrics.RecordInstanceCreation(time.Since(start))
```

Comment:
This metric is intended to measure the time it takes to create a `rlnInstance`. Here it is not doing anything useful